[EWS] Running layout tests in stress mode should skip tests marked as "Skip".

bls1999 opened the pull request:
#### 9f8b7290e181723a4d39176c60a9eb3ab317034a
<pre>
[EWS] Running layout tests in stress mode should skip tests marked as &quot;Skip&quot;.
<a href="https://bugs.webkit.org/show_bug.cgi?id=295827">https://bugs.webkit.org/show_bug.cgi?id=295827</a>
<a href="https://rdar.apple.com/155661448">rdar://155661448</a>

Reviewed by Jonathan Bedard.

When a contributor posts a PR that modifies layout tests, EWS will run those
layout tests 100 times. This is usually a great way to see if a user broke
those tests with their change; however, consider the case of a contributor
landing a change with layout tests that are skipped. This could be because that
contributor has yet to implement the feature in full (e.g., they&apos;re landing PRs
as parts of a larger feature are developed). Those layout tests still run,
despite them being skipped, because they&apos;re passed in directly on the command
line for the `run-webkit-tests` invocation.

So, this modifies the behavior of `run-layout-tests-in-stress-mode` such that
skipped tests aren&apos;t run. We achieve this by passing in `--skipped always` in
that specific step.

* Tools/CISupport/ews-build/steps.py:
    (RunWebKitTests):
        -&gt; (setLayoutTestCommand): Apply `--skipped always` based on whether we&apos;re running tests in stress mode or not.
* Tools/CISupport/ews-build/steps_unittest.py:
    (TestRunWebKitTestsInStressMode):
        -&gt; (test_success): Modify unit test.
        -&gt; (test_success_wk1): Ditto.
        -&gt; (test_failure): Ditto.
        -&gt; (test_success_additional_arguments): Ditto.
        -&gt; (test_success): Ditto.
        -&gt; (test_failure): Ditto.

Canonical link: <a href="https://commits.webkit.org/297361@main">https://commits.webkit.org/297361@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e2b22b1b4f8c7f6da279443e5b511e5be7bc2f91

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/111243 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30909 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/21372 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/117274 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/61511 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/113205 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31590 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/39491 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84552 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/114190 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/25228 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100155 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64998 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24570 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/18295 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/61094 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94607 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18365 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/120404 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/38292 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28454 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93487 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/110739 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38668 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96432 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93312 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38401 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16172 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/34277 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17968 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/38181 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37846 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/41179 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39548 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 reviewed (commented):
Thanks for the PR and thoughts. I appreciate that.

Although I wonder, isn't it good to still show the information to the PR author anyways that the tests they modified or added are failing? There shouldn't be much downside to having this EWS red.

I am afraid there might be scenarios where the PR author might assume that stress EWS will be running all the added/modified tests and a green check on stress EWS means that all those passed, while in reality, with this change, we silently don't run them all. If we analyze all such potential scenarios and find out that we are not accidentally introducing such a regression, then this change looks good.

bls1999 committed:
[EWS] Running layout tests in stress mode should skip tests marked as "Skip".
https://bugs.webkit.org/show_bug.cgi?id=295827
rdar://155661448

Reviewed by Jonathan Bedard.

When a contributor posts a PR that modifies layout tests, EWS will run those
layout tests 100 times. This is usually a great way to see if a user broke
those tests with their change; however, consider the case of a contributor
landing a change with layout tests that are skipped. This could be because that
contributor has yet to implement the feature in full (e.g., they're landing PRs
as parts of a larger feature are developed). Those layout tests still run,
despite them being skipped, because they're passed in directly on the command
line for the `run-webkit-tests` invocation.

So, this modifies the behavior of `run-layout-tests-in-stress-mode` such that
skipped tests aren't run. We achieve this by passing in `--skipped always` in
that specific step.

* Tools/CISupport/ews-build/steps.py:
    (RunWebKitTests):
        -> (setLayoutTestCommand): Apply `--skipped always` based on whether we're running tests in stress mode or not.
* Tools/CISupport/ews-build/steps_unittest.py:
    (TestRunWebKitTestsInStressMode):
        -> (test_success): Modify unit test.
        -> (test_success_wk1): Ditto.
        -> (test_failure): Ditto.
        -> (test_success_additional_arguments): Ditto.
        -> (test_success): Ditto.
        -> (test_failure): Ditto.

Canonical link: https://commits.webkit.org/297361@main
